jadx-gui/src/main/java/jadx/gui/utils/search/SearchSettings.java

```diff
@@ -0,0 +1,103 @@
+package jadx.gui.utils.search;
+
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SearchSettings {
+
+	private static final Logger LOG = LoggerFactory.getLogger(SearchSettings.class);
+
+	private String searchString;
```

Comment:
For efficiency I would recommend to make the fields final and provide for each use case an own constructor. As final fields can't change the Java code optimizer can generate faster code as several if branches can be omitted.

Reply:
I just implemented this for the searchString, ignoreCase, and useRegex option.  It should work with a single constructor as all 3 options should always have a value.